### ISSUE TITLE
Issues downloading gems from httpx://rubygems.org on Mac OS/IPv6

### ISSUE THREAD

**Issue by HoneyryderChuck** [rubygems/rubygems#2464]
I'm having a problem or would like to suggest a feature.

My current problem is downloading/installing gems, using either `gem` or `bundler`. It seems to be an https-only issue, as I could reproduce it with curl:

```
> curl https://rubygems.org
curl: (35) LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to rubygems.org:443
 openssl version
LibreSSL 2.6.4
```

It seems that the issue is network/openssl-related, as I could manage to perform curl requests for the URL when downgrading to ipv4:

```
> echo '--ipv4' >> ~/.curlrc
> curl https://rubygems.org
... (200 OK)
```

This issue is related to:

  - [x] Network problems
  - [x] Installing a library
  - [ ] Publishing a library
  - [ ] The command line `gem`
  - [ ] Other

Here are my current environment details:

```
$ gem env version
2.7.6
```

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

**Comment by segiddins** [comment]
This seems like a system configuration issue, if curl Is having the same problems. RubyGems just uses Ruby's built-in networking libraries.

**Comment by HoneyryderChuck** [comment]
TLS connection through IPv4 does work, and plain IPv6-no-TLS (if I switch source to "http://rubygems.org") too. I find this odd, although I'd expect to have found more similar complaints. Do you connect over IPv6 to download gems in a Mac?

**Comment by colby-swandale** [comment]
I'm not sure how IPv6 without TLS works but with TLS doesn't. I suggest running [Fastly's debugger](https://www.fastly-debug.com/) and contacting their support team to help resolve your issue.

**Comment by HoneyryderChuck** [comment]
Apparently I ran into this issue: https://github.com/pypa/pip/issues/5374 . This might be good information to have.

**Comment by sparkjerry** [comment]
An error occurred while installing nokogiri (1.8.5), and Bundler cannot
continue.
Make sure that `gem install nokogiri -v '1.8.5' --source
'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  factory_bot_rails was resolved to 4.11.1, which depends on
    railties was resolved to 4.2.10, which depends on
      actionpack was resolved to 4.2.10, which depends on
        actionview was resolved to 4.2.10, which depends on
          rails-dom-testing was resolved to 1.0.9, which depends on
            nokogiri
please how do i dealt with this issue plsssssssssssssssssssssssss

**Comment by bronzdoc** [comment]
Closing since this is not a Rubygems issue.
